### PR TITLE
Added disabled count per groups & fixed disabled users query 

### LINKED ISSUE #8008: User count for groups with disabled users is wrong
* having a group "A" with two users "user1" and "user2"
* disable "user1"
* expected: group count for group "A" is 2 and shows the disabled user or it is 1 and doesn't show the user
* actual: group count is 2 and doesn't show the user

bildschirmfoto 2018-01-23 um 15 17 38
bildschirmfoto 2018-01-23 um 15 17 43

cc @michag86 as you also experience this

cc @blizzz @nickvergessen @rullzer

### LINKED ISSUE #8009: Disabled users count for sub admins wrong
* having a group "A" with two users "user1" and "user2"
* having users "user3" and "user4" without a group
* disable "user1" and "user3"
* make "user2" the group admin of group "A"
* login as "user2"
* expected: disabled count is 1 (because only one of the users that are in the scope of the group admin "user2" is disabled)
* actual: disabled count is 2 (user1 and user3)

admin view:
bildschirmfoto 2018-01-23 um 15 46 53

bildschirmfoto 2018-01-23 um 15 46 55

subadmin view:
bildschirmfoto 2018-01-23 um 15 46 57

cc @michag86 for noticing this

@nickvergessen @rullzer @danxuliu @blizzz for the user management 

May be related to #8008

### PR DESCRIPTION
**Pull request by skjnldsv**
After #9261
Fix #8009
Fix #8008
I feel like it's safer to return a different entry instead of directly subtracting the disabled users to the total user count. People might want to get the full data anyway.

### CODE REVIEW

File: lib/private/Group/Database.php [hunk 1]
```diff
@@ -53,6 +56,7 @@
  */
 class Database extends ABackend
 	implements IAddToGroupBackend,
+			   ICountDisabledInGroup,
```
Comment (MorrisJobke) [inline lib/private/Group/Database.php]: Indentation is a bit off ;)

**Comment by MorrisJobke** [comment]
@michag86 Mind to test this?

File: lib/private/Group/Database.php (hunk 1 above)
Comment (skjnldsv) [reply lib/private/Group/Database.php]: Hum, is it really?

**Comment by MorrisJobke** [comment]
Tested and works nice. There is only one thing left:

* [ ] count in left sidebar is not updated

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

File: apps/provisioning_api/lib/Controller/GroupsController.php [hunk 2]
```diff
@@ -111,7 +111,12 @@ public function getGroupsDetails(string $search = '', int $limit = null, int $of
 		$groups = $this->groupManager->search($search, $limit, $offset);
 		$groups = array_map(function($group) {
 			/** @var IGroup $group */
-			return ['id' => $group->getGID(), 'displayname' => $group->getDisplayName()];
+			return [
+				'id' => $group->getGID(),
+				'displayname' => $group->getDisplayName(),
+				'usercount' => $group->count(),
```
Comment (blizzz) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: can be expensive, that's why we disabled counting for external users.

Comment (blizzz) [reply apps/provisioning_api/lib/Controller/GroupsController.php]: > @blizzz this is the extended groups request, people can still use the traditionnal getGroups request if they need to?

:+1: , that's fine then. 

> Does ldap provide a count implementation?

yes, it implements the standard count routines.

Comment (rullzer) [reply apps/provisioning_api/lib/Controller/GroupsController.php]: @blizzz so you are good with this?

**Comment by skjnldsv** [comment]
Okay, it's done!

@michag86 can you test again? :)

**Comment by skjnldsv** [comment]
**Please wait before merging, there is something I need to test before. :)**

**Comment by skjnldsv** [comment]
Failure unrelated.
@nextcloud/designers

**Comment by blizzz** [comment]
@skjnldsv we had some discussions lately (incl. opt-in user counting) and you also asked for some output related to LDAP. What was the outcome?

**Comment by skjnldsv** [comment]
@blizzz I want to add a specific override for ldap.
Right now we have some hardcoded values for the ldap plugin but that make no real sense since it's an external app.

I'm suggesting we add a specific config to disable the user count (and enabling the ldap app will automatically activate this option). Like this we have the count by default for all our backend but an admin ca decide to disable it if he thinks it's drawing too much perfs :)
What do you think?

**Comment by blizzz** [comment]
Yeah, better than a hardcoded string.

**Comment by skjnldsv** [comment]
@blizzz let's get this in then :)

File: lib/private/Group/Database.php (hunk 1 above)
Comment (blizzz) [reply lib/private/Group/Database.php]: appears to be off, yes

Comment (skjnldsv) [reply lib/private/Group/Database.php]: Strange. Fixed now. :)
capture d ecran_2018-05-25_14-42-39

**Comment by blizzz** [comment]
@skjnldsv can you make the 0s go away? The old way was if the count is 0 don't show any number. and this applies with LDAP enabled.

screenshot_20180525_145231

**Comment by skjnldsv** [comment]
Makes sense! :)
Incoming!

**Review by blizzz — APPROVED** [review]

**Pull request merged**